### ISSUE TITLE
DD_TAGS with spaces inconsistent with dd-trace-java agent expecting commas

### ISSUE THREAD

**Issue by drobert** [DataDog/heroku-buildpack-datadog#187]
Using the latest heroku buildpack with the latest dd-java-agent (0.52.0) there seems to be a disagreement on the proper format of DD_TAGS.

In buildpack commit https://github.com/DataDog/heroku-buildpack-datadog/pull/182 it seems DD_TAGS have been standardized to be space-separated. 

However with the latest dd-java-agent this is output in the application (JVM) logs:
> datadog.trace.api.Config - Invalid config for dd.tags: 'dyno:somedyno.2 dynotype:somedyno buildpackversion:dev appname:redacted env:production'. Must match 'key1:value1,key2:value2'.

In my setup, we are not setting a `-Ddd.tags` value directly, it is being parsed from DD_TAGS

**Comment by arapulido** [comment]
Hello @drobert ! Both comma separated and space separated (or a combination of both) should be accepted by the Heroku buildpack. This is mentioned on the documentation: 
```
(Note: comma-separated string in buildpack versions 1.16 and earlier; this is still supported to maintain backward compatibility)
```

Can you try adding your tags in `DD_TAGS` separating them by comma, please? It should work even after buildpack 1.16

**Comment by arapulido** [comment]
> @arapulido what I'm noting is that the buildpack itself is _setting_ DD_TAGS using spaces, while the java agent attached to our JVM is expecting this value to be comma separated. I'm not setting DD_TAGS myself anywhere. I'm ticketing an inconsistency between two of datadog's offerings: this and https://github.com/DataDog/dd-trace-java

Ah, indeed. `dd-trace-java` should accept space-separated values. I will follow this up to fix it on their side. In the mean time, you can go back to buildpack version 1.16 while this is getting fixed. 

I will keep this issue open to track this. 

Thanks for bringing this to our attention!

**Comment by arapulido** [comment]
I am reopening this, as we needed to revert the workaround

**Comment by arapulido** [comment]
@drobert Hello Daniel. This has now been fixed in the java dd tracer in the 0.66.0 release: https://github.com/DataDog/dd-trace-java/releases/tag/v0.66.0

If you upgrade the tracing library, it will fix the issue. Closing this now.

Please, reopen if you still find an issue after upgrading the tracing library

**Comment by benrobster** [comment]
We're seeing this issue with ddtrace-py running in conjunction with the heroku buildpack as well... 

based on the above path that resolving it for java took, I assume the path forward is to work with the ddtrace-py team to get it patched to support space-separated tags? 
 https://github.com/DataDog/dd-trace-py/blob/3a9f1216275e7b081cbc3b247343558e7709920d/ddtrace/utils/formats.py#L87 
cc @colinnutrislice
